tests/test_GameWorld.cpp

```diff
@@ -16,6 +16,59 @@ BOOST_AUTO_TEST_CASE(test_gameobject_id)
 
 	BOOST_CHECK_NE( object1->getGameObjectID(), object2->getGameObjectID() );
 }
+
+BOOST_AUTO_TEST_CASE(test_offsetgametime)
+{
+	GameWorld gw(&Global::get().log, &Global::get().work, Global::get().d);
+	gw.state = new GameState();
+
+	BOOST_CHECK_EQUAL(0, gw.getHour());
+	BOOST_CHECK_EQUAL(0, gw.getMinute());
+
+	gw.offsetGameTime(30);
+	BOOST_CHECK_EQUAL(0, gw.getHour());
+	BOOST_CHECK_EQUAL(30, gw.getMinute());
+
+	gw.offsetGameTime(30);
+	BOOST_CHECK_EQUAL(1, gw.getHour());
+	BOOST_CHECK_EQUAL(0, gw.getMinute());
+
+	gw.offsetGameTime(-30);
+	BOOST_CHECK_EQUAL(0, gw.getHour());
+	BOOST_CHECK_EQUAL(30, gw.getMinute());
+
+	gw.offsetGameTime(-60);
+	BOOST_CHECK_EQUAL(23, gw.getHour());
+	BOOST_CHECK_EQUAL(30, gw.getMinute());
+
+	gw.offsetGameTime(30);
+	BOOST_CHECK_EQUAL(0, gw.getHour());
+	BOOST_CHECK_EQUAL(0, gw.getMinute());
+
+	gw.offsetGameTime(24*60);
+	BOOST_CHECK_EQUAL(0, gw.getHour());
+	BOOST_CHECK_EQUAL(0, gw.getMinute());
+
+	gw.offsetGameTime(8*60 + 25);
```

Comment:
Note that this is the only test case which goes forward by non-30 minute increments.
As we have 2 directions (+ and -) this is not enough to test all cases (do we know that -25 works?).

We should have more cases which give a remainder if 60 is divided by the minutes and we should have that in both directions (+ and -). I didn't think about this very much tho, so maybe it's useless.